piglr/game.py

```diff
@@ -75,16 +116,37 @@ def reset(self) -> Union[dict, None]:
         return self.state, None
 
     def roll(self) -> Union[dict, None]:
-        d = random.randint(1, self._state.dn)
-        if d == 1:
-            self._increment_turn(bank=False)
-        else:
-            self._state.rolls += 1
-            self._state.bank += d
+        """Roll dice.
+
+        Rolls dice for current player.
+
+        Returns:
+            - State property :attr:`piglr.game.Game.state`.
+            - :obj:`None`.
+
+        """
+        for _ in range(self._state.num_dice):
+            d = random.randint(1, self._state.dn)
+            if d == 1:
```

Comment:
I think we should only check for whether or not the total sum for the roll of all dice is equal to the number of dice. e.g. If we're playing with 2 dice then it's ok to roll a `1` on 1 die as long as you don't roll a `1` on the other die.

```python
roll_value = `sum(random.randint(1,` self._state.dn)) for _ in range(self._state.num_dice))
if roll_value == self._state.num_dice:
    self._increment_turn(bank=False)
    return self.state, None
```

Reply:
Let's check the official game rules. I was under the impression a single 1 was enough to bust your roll.

Reply:
I thought that was mostly because they were only using 1 die in that numberphile video

Reply:
I'm seeing different rules, but most of the instructions I've seen mention only 1 die. https://nzmaths.co.nz/resource/greedy-pig.

If we're trying to accommodate multiple dice, then I'd say that we would need to check if the value of the roll is the number of die involved.

Reply:
Closing the loop, we chatted and agreed on the one die landing on 1 implementation.